packages/vue/src/pinia.ts

```diff
@@ -0,0 +1,82 @@
+import { addBreadcrumb, getClient, getCurrentScope, getGlobalScope } from '@sentry/core';
+import { addNonEnumerableProperty } from '@sentry/utils';
+
+// Inline PiniaPlugin type
+type PiniaPlugin = (context: {
+  store: {
+    $state: unknown;
+    $onAction: (callback: (context: { name: string; after: (callback: () => void) => void }) => void) => void;
+  };
+}) => void;
+
+type SentryPiniaPluginOptions = {
+  attachPiniaState?: boolean;
+  actionTransformer: (action: any) => any;
+  stateTransformer: (state: any) => any;
+};
+
+export const createSentryPiniaPlugin: (options?: SentryPiniaPluginOptions) => PiniaPlugin = (
+  options: SentryPiniaPluginOptions = {
+    attachPiniaState: true,
+    actionTransformer: action => action,
+    stateTransformer: state => state,
+  },
+) => {
+  const plugin: PiniaPlugin = ({ store }) => {
+    options.attachPiniaState &&
+      getGlobalScope().addEventProcessor((event, hint) => {
+        try {
+          hint.attachments = [
+            ...(hint.attachments || []),
+            {
+              filename: 'pinia_state.json',
+              data: JSON.stringify(store.$state),
```

Comment:
L: Can there be multiple files (because the linked event has two)? Users could benefit from having a timestamp in the filename as there can be multiple files in the attachments. Something like `hh:mm_pinia_state` would be enough.

Reply:
Looks like these two are two different stores. So, I updated the file name format to include both the store id and the timestamp. [Sample Event](https://sentry-sdks.sentry.io/issues/5939879614/events/9c561a7e36504fb3b069886d7ebb1da1/)